docs/topics/native/native-objc-interop.md

```diff
@@ -569,6 +573,56 @@ binaries.framework {
 }
 ```
 
+### Forward declarations
+
+To import forward declarations, use the `objcnames` package. For example, to import a `objcstructName` forward declaration
+declared in an Objective-C library with a `library.package`, use a special forward declaration package:
+`import objcnames.structs.objcstructName`.
+
+To transfer objects between libraries, you need to make an explicit `as` cast to and from the corresponding Objective-C
+forward declaration. Consider two objcinterop libraries, one that uses `objcnames.protocols.ForwardDeclaredProtocolProtocol`
+and the other that has an actual definition:
+
+```ObjC
+// First objcinterop library
+#import <Foundation/Foundation.h>
+
+@protocol ForwardDeclaredProtocol;
+
+NSString* consumeProtocol(id<ForwardDeclaredProtocol> s) {
+    return [NSString stringWithUTF8String:"Protocol"];
+}
+```
+
+```ObjC
+// Second objcinterop library
+// Header:
+#import <Foundation/Foundation.h>
+@protocol ForwardDeclaredProtocol
+@end
+// Implementation:
+@implementation ForwardDeclaredProtocolImpl : NSObject
```

Comment:
It should be `@interface ForwardDeclaredProtocolImpl : NSObject <ForwardDeclaredProtocol>` instead.

Reply:
Thank you! It's good that we decided to add this because the Obj-C example was copied from the compatibility guide, so we had an error there too. fixed both